ocs_ci/deployment/ibmcloud.py

```diff
@@ -219,17 +226,24 @@ def get_release_image(self):
             if "release image" in line:
                 return line.split(" ")[2].strip()
 
-    def get_resource_group(self):
+    def get_resource_group(self, return_id=False):
         """
         Retrieve and set the resource group being utilized for the cluster assets.
+
+        Args:
+            return_id (bool): If True, it will return ID instead of name.
+
+        Returns:
+            str: name or ID of resource group if found.
+            None: in case no RG found.
+
         """
         cmd = "ibmcloud resource groups --output json"
         proc = exec_cmd(cmd)
         logger.info("Retrieving cluster resource group")
         resource_data = json.loads(proc.stdout)
         for group in resource_data:
-            if group["name"].startswith(self.cluster_name):
-                # TODO: error prone if cluster_name is a substring of another cluster
+            if group["name"].startswith(f"{self.cluster_name}-"):
```

Comment:
This might be still error prone for cluster name which is a substring of another cluster.
The random suffix have constant length, doesn't have? If yes, we could remove it from the end of the group name and then compare it with the cluster name.

Reply:
good point, updated